storage/innobase/buf/buf0buf.cc

```diff
@@ -921,23 +923,17 @@ void mem_pressure::pressure_routine(mem_pressure *m)
         }
       }
 
+#ifdef UNIV_DEBUG
       if (p.revents & POLLIN)
       {
-        char c= '\0';
-        /* signal to quit */
-        if (read(p.fd, &c, 1) >=0)
-          switch (c) {
-          case 'Q':
-            goto shutdown;
-#ifdef UNIV_DEBUG
-          case 'G':
-            buf_pool.garbage_collect();
-#endif
-          }
+        uint32 u;
+        /* we haven't aborted, so this must be a debug trigger */
+        if (read(p.fd, &u, sizeof(u)) >=0)
+          buf_pool.garbage_collect();
```

Comment:
We wrote `uint64_t` but are reading only `uint32`. Could this be a single-byte read and write?

Reply:
eventfd prescribes 64 bit